### PR TITLE
convert event-related timestamps to new system

### PR DESCRIPTION
**Pull request by Goober5000**
This change covers just the timestamps associated with goals and events, for better review and isolation purposes.

Follow-up to #4569.

### CODE REVIEW

**Comment by Goober5000** [comment]
After extensive testing and fixing several bugs, this is now ready for review.

File: code/mission/missiongoals.cpp [hunk 1]
```diff
@@ -1522,3 +1551,23 @@ int mission_event_find_sexp_tree(int root_node)
...
+
+int mission_event_lookup(const char *name)
+{
+	Assertion(name != nullptr, "Event name cannot be null!");
+
+	for (int i = 0; i < Num_mission_events; ++i)
+		if (!stricmp(Mission_events[i].name, name))
+			return i;
+	return -1;
+}
```
Comment (BMagnu) [inline code/mission/missiongoals.cpp:1573]: What are these added for?
Remnants of a different PR maybe?

**Review by BMagnu — CHANGES_REQUESTED** [review]
Looks mostly good.
As this is fairly deep in the logic, it's gonna need a lot of testing, but it sounds like you did that.
This interval behaviour will need to be well-documented on the respective wiki page as well.
One question though

File: code/mission/missiongoals.cpp (hunk 1 above)
Comment (Goober5000) [reply code/mission/missiongoals.cpp:1573]: These are in preparation for #4634, but I can move them to that PR if you think I should.

Comment (BMagnu) [reply code/mission/missiongoals.cpp:1573]: eh, its fine in that case. I was just wondering

**Pull request merged**